### DISCUSSION TITLE
Any downsides/concerns with turning on Sidekiq.transactional_push! / Sidekiq::TransactionAwareClient

### DISCUSSION THREAD

**Discussion by oskarpearson** [sidekiq/sidekiq#5725] [Q&A] [answered]
Hi All

We've a significantly-sized application that makes extensive use of Sidekiq. Right now we've got some custom code that uses something like https://juanitofatas.com/series/rails/active-record-transaction to delay jobs until after the transaction commits.

I'd like to remove this over time, and `Sidekiq.transactional_push!` is an obvious place to start.

I was wondering:

- Are people finding it safe in production? I'm aware of https://github.com/mperham/sidekiq/discussions/5672 which seems positive
- Any gotchas? We don't use `push_bulk`
- Should we consider using `sidekiq_options client_class: Sidekiq::TransactionAwareClient` worker-by-worker? Or would people just turn on transactional_push generally?

I'm not seeing much documentation on it - and things like https://github.com/mperham/sidekiq/wiki/FAQ#why-am-i-seeing-a-lot-of-cant-find-modelname-with-id12345-errors-with-sidekiq don't refer to it.

Thanks!

**Comment by mperham** [comment, marked as answer]
I'm not aware of any issues and very large customers like Gusto have been using it for months now (and submitting improvements).

**Reply by oskarpearson** [reply]
Thanks. Just wanted to give another vote of confidence here. We (237,647,853 jobs processed since we last cleared stats) turned it on and have had no issues at all.

**Comment by stevenharman** [comment]
I too am enabling `transactional_push!` and got some questions/comments from co-workers asking what this was, confused b/c they couldn't find documentation on it, wondering about known-downsides, etc… I ended up here while searching for some docs to point them to

Which leaves me wondering - are there existing docs and I'm just missing them? Or is this something we should add to the Wiki? If so, as a new section of the [Reliability page](https://github.com/sidekiq/sidekiq/wiki/Reliability)?

**Reply by mperham** [reply]
Yeah, it was released as a sort of beta feature in 6.5 and never got properly documented. Your suggestion sounds perfect.

**Comment by adamniedzielski** [comment]
@mperham I wanted to improve the wiki on this topic, but it looks like I don't have the necessary access rights.

Feel free to apply this patch:

[0001-Improve-documentation-of-Transaction-Aware-Client.patch](https://github.com/sidekiq/sidekiq/files/13785763/0001-Improve-documentation-of-Transaction-Aware-Client.patch)
